gym/spaces/multi_discrete.py

```diff
@@ -8,6 +8,7 @@
 from gym.spaces.space import Space
 from gym.utils import seeding
 
+RECURSIVE_LIST_INT = List[Union["RECURSIVE_LIST_INT", int]]
```

Comment:
Does this actually work well with type checkers? I see there's a similar pattern in the existing code that I never noticed, but last I checked, python didn't really like that.

Reply:
Pycharm doesn't seem to interpret it correctly so Im not sure but this should be the correct typing. 
Any suggestions?

Reply:
My understanding is that (a) it's valid in python, (b) mypy doesn't support it, (c) pyright supports it (hence no errors in CI)

I'd refrain from using this since it will cause issues for at least some users (myself included), at least until recursive types are more widely supported.

The simple alternative is just an unspecified `List`

Reply:
Agreed, I will change it to list